apps/prairielearn/python/prairielearn/timeouts.py

```diff
@@ -0,0 +1,161 @@
+"""Utilities for performing local, threaded timeouts.
+Implemenation from https://github.com/glenfant/stopit under the MIT license.
+
+```python
+from prairielearn.timeouts import ...
+```
+"""
+
+import ctypes
```

Comment:
A test case would be good.

Reply:
+1 on this. When I first tried implementing this, I ran into issues where it would have trouble timing out internally without hitting the external time limit for executing python code. You should also make sure to test locally with at least one large expression.